src/common/FrameManager.ts

```diff
@@ -454,16 +455,16 @@ export class Frame {
 
   async $eval<ReturnType extends any>(
     selector: string,
-    pageFunction: Function | string,
-    ...args: unknown[]
+    pageFunction: EvaluateFn | string,
+    ...args: SerializableOrJSHandle[]
```

Comment:
What would be really helpful here (more than trying to constrain the serializable types) would be to make the arguments and return type type-safe, like this:
```ts
$eval<A extends any[], R>(pageFunction: (...args: A) => R, ...args: A): Promise<R>
```